src/Storages/MergeTree/DataPartStorageOnDisk.cpp

```diff
@@ -768,41 +772,43 @@ void DataPartStorageOnDisk::changeRootPath(const std::string & from_root, const
     root_path = to_root.substr(0, dst_size) + root_path.substr(prefix_size);
 }
 
-DataPartStorageBuilderOnDisk::DataPartStorageBuilderOnDisk(VolumePtr volume_, std::string root_path_, std::string part_dir_)
-    : volume(std::move(volume_)), root_path(std::move(root_path_)), part_dir(std::move(part_dir_))
-{
-}
-
-std::unique_ptr<ReadBufferFromFileBase> DataPartStorageBuilderOnDisk::readFile(
```

Comment:
Why not to leave reading?

Reply:
Because this class is intended to be "write-once" part creator... However, this initial idea almost failed in the current implementation.